src/helpers/request.ts

```diff
@@ -0,0 +1,77 @@
+import config from '../config';
+import queryString from 'qs';
+import { getCookie } from '../helpers/cookieHelper';
+
+function getEndpointFullUrl(api: string, queryParams: any = {}): string {
+  return `${config.backendUrl}/${api}?${queryString.stringify(queryParams, { arrayFormat: 'brackets' })}`;
+}
+
+export const get = async (url: string, queryParams = {}) => {
+  const response = await fetch(getEndpointFullUrl(url, queryParams), {
+    headers: {
+      'Content-Type': 'application/json',
+      'auth-token': getCookie(config.tokenKey),
+    },
+  });
+
+  return response.json();
+};
+
+export const post = async (url: string, data = {}) => {
+  // Default options are marked with *
+  const response = await fetch(getEndpointFullUrl(url), {
+    method: 'POST',
+    mode: 'cors',
+    cache: 'no-cache',
+    credentials: 'same-origin',
+    headers: {
+      'Content-Type': 'application/json',
+      'auth-token': getCookie(config.tokenKey),
+    },
+    body: JSON.stringify(data),
+  });
+
+  try {
+    checkStatus(response);
+    return parseJSON(response);
+  } catch (err: any) {
+    const message = await err.response.json();
+    throw message;
+  }
+};
+
+export const postBeaconReq = async (url: string, data = {}) => {
```

Comment:
Do we need this?

Reply:
yes.. when user refreshes the page or close the tab while configuring the quiz, I'm storing the data in BE